docs/source/ReferenceGuide.rst

```diff
@@ -361,24 +361,80 @@ The instances should have:
 
 .. [#f1] Avocado plugins can introduce additional test types.
 
-Pre and post tests plugins
-==========================
+Pre and post plugins
+====================
+
+Avocado provides interfaces with which custom plugins can register to
+be called at various times.  For instance, it's possible to trigger
+custom actions before and after the execution of a :class:`job
+<avocado.core.job.Job>`, or before and after the execution of the
+tests from a job :data:`test suite <avocado.core.job.Job.test_suite>`.
+
+Let's discuss each interface briefly.
+
+Before and after jobs
+---------------------
 
 Avocado supports plug-ins which are (guaranteed to be) executed before the
 first test and after all tests finished. The interfaces are
-:class:`avocado.core.plugin_interfaces.JobPre`, resp.
-:class:`avocado.core.plugin_interfaces.JobPost`.
-
-Note the ``pre_tests`` might not be executed due to earlier failure which
-prevents the tests from being executed.
-
-The same applies for ``post_tests``, but it is possible to have ``post_tests``
-executed even when ``pre_tests`` were not. Additionally the ``post_tests``
-are (obviously) not executed on ``SIGKILL``. On the other hand they are
-executed on ``SIGTERM`` and ``KeyboardInterrupt`` while running
-the tests, but once the ``post_tests`` are executed the ``KeyboardInterrupt``
-or ``SystemExit`` interrupts their processing (to avoid hangs) and remaining
-plug-ins will **NOT** be executed.
+:class:`avocado.core.plugin_interfaces.JobPre` and
+:class:`avocado.core.plugin_interfaces.JobPost`, respectively.
+
+The :meth:`pre <avocado.core.plugin_interfaces.JobPre.pre>` method of
+each installed plugin of type ``job.prepost`` is called by the ``run``
+command, that is, anytime an ``avocado run <valid_test_reference>``
+command is executed.
+
+Then, immediately after that, the job's :meth:`run
+<avocado.core.job.Job.run>` method is called, which attempts to run
+all job phases, from test suite creation to test execution.
+
+No matter what happens up to this point, as long as the Avocado test
+runner is still in execution (that is, has not been terminated by a
+system condition that it can not control), it will attempt to run the
```

Comment:
`that it can not control or unless SystemExit or KeyboardInterrupt during post stage is issued)`

... we can control these but we interrupt the execution on them.